shell/src/main/java/alluxio/cli/bundler/CollectInfo.java

```diff
@@ -0,0 +1,177 @@
+public class CollectInfo extends AbstractShell {
+  protected static final String TARBALL_NAME = "alluxio-info.tar.gz";
+
+  private static final Map<String, String[]> CMD_ALIAS = ImmutableMap.of();
+
+  // In order for a warning to be displayed for an unstable alias, it must also exist within the
+  // CMD_ALIAS map.
+  private static final Set<String> UNSTABLE_ALIAS = ImmutableSet.of();
+
+  /**
+   * Main method, starts a new CollectInfo.
+   *
+   * @param argv array of arguments given by the user's input from the terminal
+   */
+  public static void main(String[] argv) throws IOException, InvalidArgumentException {
+    int ret = 0;
+
+    InstancedConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
+
+    // Execute the Collectors one by one
+    // Reduce the RPC retry max duration to fall earlier for CLIs
+    conf.set(PropertyKey.USER_RPC_RETRY_MAX_DURATION, "5s", Source.DEFAULT);
+    CollectInfo shell = new CollectInfo(conf);
+
+    // If the args are not valid, return early
+    // TODO(jiacheng): catch in CollectInfoAll
+    if (argv.length < 2) {
+      shell.printUsage();
+      throw new InvalidArgumentException(ExceptionMessage.INVALID_ARGS_NUM_INSUFFICIENT
+              .getMessage("collectInfo", 2, argv.length));
+    }
+
+    // Determine the command and working dir path
+    String subCommand = argv[0];
+    // TODO(jiacheng): phase 2 get targetDirPath from parsed command
+    String targetDirPath = argv[1];
+
+    // There are 2 cases:
+    // 1. Execute "all" commands
+    // 2. Execute a single command
+    List<File> filesToCollect = new ArrayList<>();
+    if (subCommand.equals("all")) {
+      // Case 1. Execute "all" commands
+      System.out.println("Execute all child commands");
+      for (Command cmd : shell.getCommands()) {
+        System.out.format("Executing %s", cmd.getCommandName());
+
+        // Replace the action with the command to execute
+        // TODO(jiacheng): phase 2 handle argv difference?
+        String[] replacementArgv =
+                (String[]) ArrayUtils.addAll(new String[]{cmd.getCommandName()},
+                        ArrayUtils.subarray(argv, 1, argv.length));
+        int childRet = shell.executeAndAddFile(replacementArgv, filesToCollect);
+
+        // If any of the commands failed, treat as failed
+        if (ret == 0 && childRet != 0) {
+          ret = childRet;
+        }
+      }
+    } else {
+      // Case 2. Execute a single command
+      int childRet = shell.executeAndAddFile(argv, filesToCollect);
+      if (ret == 0 && childRet != 0) {
+        ret = childRet;
+      }
+    }
+
+    // TODO(jiacheng): phase 2 add an option to disable bundle
+    // Generate bundle
+    System.out.format("Archiving dir %s", targetDirPath);
+    String tarballPath = Paths.get(targetDirPath, TARBALL_NAME).toAbsolutePath().toString();
+    if (filesToCollect.size() == 0) {
+      System.err.format("No files to add. Tarball %s will be empty!", tarballPath);
```

Comment:
Why do we need to generate an empty tarball?

Reply:
On 2nd thought I think I was wrong, an empty tarball shouldn't be generated. At least 1 operation will be executed and at least 1 files will be generated. So having no files to tarball implies an error state.

Reply:
If no file is generated, then the empty tarball contains no useful information. Should we just print error message and not generate any tarball?